### PR TITLE
Allow the consistent selection of a GPU device by DRM ID

### PR DESCRIPTION
**Pull request by roosterfish**
Allow adding physical GPU devices to containers by setting the DRM ID with the `id` setting. Before a new device `/dev/nvidia[0123...]` was added for each graphics card regardless of what was configured with `id`. This was caused by a filtering mechanism that only took `vendorid`, `pci` and `productid` into account.

A new generic filter function is now used for each GPU device type (containers & virtual machines) which is checking all the attributes `vendorid`, `pci`, `productid` and `id`. 

For `gputype=mig` the check for the `id` filter is now also added since it was missing but supported according to the docs.

Fixes https://github.com/lxc/lxd/issues/11442

### CODE REVIEW

File: lxd/device/device_utils_generic.go
Comment (tomponline) [inline lxd/device/device_utils_generic.go:15]: "if no device" would be better as "if empty address supplied"

**Review by tomponline — APPROVED** [review]
LGTM! But @stgraber knows more about the GPU pass-through stuff, so I'll let him take a look before merging.

File: lxd/device/device_utils_generic.go
Comment (roosterfish) [reply lxd/device/device_utils_generic.go]: True, I have rebased the commit with an updated description.

**Pull request merged**
